pkg/rainbow/api/rest.go

```diff
@@ -0,0 +1,290 @@
+// Copyright (c) 2022 Teal.Finance
+//
+// Use of this source code is governed by an MIT-style
+// license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+package api
+
+import (
+	"encoding/csv"
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"strings"
+
+	"github.com/go-chi/chi/v5"
+	"github.com/mpvl/unique"
+
+	"github.com/teal-finance/rainbow/pkg/rainbow"
+)
+
+func (h APIHandler) Options(w http.ResponseWriter, r *http.Request) {
+	sa, format := query(r)
+
+	options, _ := h.Service.Options(sa)
+	if len(options) == 0 {
+		http.Error(w, "No Content", http.StatusNoContent)
+
+		return
+	}
+
+	err := h.writeOptions(w, options, sa, format)
+	if err != nil {
+		log.Printf("WRN writeOptions options=%v sa=%v format=%v err=%v", len(options), sa, format, err)
+	}
+}
+
+func (h APIHandler) writeOptions(w http.ResponseWriter, options []rainbow.Option, sa rainbow.StoreArgs, format string) error {
+	switch {
+	case format == "": // The most frequent first
+		return h.replyJSON(w, options)
+
+	// We use strings.Contains() because format can be
+	// "text/csv" when provided by the "Accept" header.
+	// This looks stupid but it's just simple.
+	case strings.Contains(format, "csv"):
+		setFilename(w, sa, ".csv")
+		return h.replyCSV(w, options, ',')
+
+	case strings.Contains(format, "tsv"):
+		setFilename(w, sa, ".tsv")
+		return h.replyCSV(w, options, '\t')
+
+	case strings.Contains(format, "json"):
+		setFilename(w, sa, ".json")
+		fallthrough
+
+	default:
+		return h.replyJSON(w, options)
+	}
+}
+
+func (h APIHandler) replyJSON(w http.ResponseWriter, options []rainbow.Option) error {
+	w.Header().Set("Content-Type", "application/json")
+
+	if err := json.NewEncoder(w).Encode(options); err != nil {
+		http.Error(w, "INTERNAL_SERVER_ERROR", http.StatusInternalServerError)
+
+		return err
+	}
+
+	return nil
+}
+
+func (h APIHandler) replyCSV(w http.ResponseWriter, options []rainbow.Option, comma rune) error {
+	// Write in CSV format.
+	csvWriter := csv.NewWriter(w)
+	csvWriter.Comma = comma
+
+	w.Header().Set("Content-Type", "text/csv")
+
+	// Header
+	row := []string{
+		/* 00 */ "Name",
+		/* 01 */ "Expiry",
+		/* 02 */ "Asset",
+		/* 03 */ "Currency",
+		/* 04 */ "Strike",
+		/* 05 */ "Type",
+		/* 06 */ "Provider",
+		/* 07 */ "Layer",
+		/* 08 */ "Chain", // if empty --> ExchangeType
+
+		/* 09 */ "Bid Price #1",
+		/* 10 */ "Bid Size #1",
+		/* 11 */ "Bid Price #2",
+		/* 12 */ "Bid Size #2",
+		/* 13 */ "Ask Price #1",
+		/* 14 */ "Ask Size #1",
+		/* 15 */ "Ask Price #2",
+		/* 16 */ "Ask Size #2",
```

Comment:
Does the number ascommentaries are that usefull ?

Reply:
These numbers are the indexes used just after in the source code.
This is useful to flatten the `Option` structure.

Reply:
Should staty with golang format then (and maybe some comment to explain their goal.

```go
"Ask Price #1" // 13
```
```

Reply:
Good idea :+1: I did it and I improved again the API documentation within the README.md
https://github.com/teal-finance/rainbow/tree/restore-rest-api#api